### ISSUE TITLE
To continue a discussion I need to keep starting off with "Based on my notes"

### ISSUE THREAD

**Issue by farzanmaghami** [brianpetro/obsidian-smart-connections#407]
It's not a great UX having to start every response that requires analysis of my notes in an existing conversation with "Based on my notes".

It makes sense to start the conversation this way, but for follow-up conversations, it's not natural to have to write this way. See the screenshot for an example of this.

[SC_2024-01-10_16-21-03.pdf](https://github.com/brianpetro/obsidian-smart-connections/files/13883399/SC_2024-01-10_16-21-03.pdf)

**Comment by huachuman** [comment]
Yes, this is a serious annoyance. There must be a way around it.

**Comment by brianpetro** [comment]
@farzanmaghami @huachuman thanks for the feedback

My plan is to have major updates to the chat in version 2.1.

One of those updates I'm considering is using the AI to detect whether looking up notes is necessary, while that process is currently triggered by phrases like "based on my notes."

There's also the context limit issue that currently prevents a lot of previous context, for example the retrieved notes, from being included in context for subsequent messages. In v2.1 the chat history will be saved in a markdown format that Smart Connections can use for partial retrieval, similar to how notes are retrieved. 

Do you have any thoughts on that? Does it make sense? Seem like what you're aiming at?

🌴

**Comment by ineation** [comment]
@brianpetro have you considered using OpenAI assistant API? It includes smart mechanisms (ie summarization + truncation) to deal with the context window limitations. It also provides an easy way to access to all the cool capabilities.

**Comment by brianpetro** [comment]
@ineation 

I have, but currently have decided against it because it would 1) lock-in users to OpenAI services, which goes against one of my v2 objectives of enabling more service/model options. And 2) by my understanding the assistant API would come with a substantial cost increase due to having to pay monthly for any data stored on OpenAI servers.

I think the [Smart Connections GPT](https://chat.openai.com/g/g-SlDDp07bm-smart-connections-for-obsidian) might be a good option if you're ok with the above two points.

Thanks for interest in making Smart Connections the best it can be 🤓

🌴
